app/plugin_key_value_store.go

```diff
@@ -45,18 +44,27 @@ func (a *App) SetPluginKeyWithExpiry(pluginId string, key string, value []byte,
 }
 
 func (a *App) GetPluginKey(pluginId string, key string) ([]byte, *model.AppError) {
-	result := <-a.Srv.Store.Plugin().Get(pluginId, getKeyHash(key))
+	result := <-a.Srv.Store.Plugin().Get(pluginId, key)
+
+	if result.Err == nil {
+		kv := result.Data.(*model.PluginKeyValue)
+		return kv.Value, nil
+	} else if result.Err != nil {
+		if result.Err.StatusCode != http.StatusNotFound {
+			mlog.Error(result.Err.Error())
+			return nil, result.Err
+		}
+	}
 
+	result = <-a.Srv.Store.Plugin().Get(pluginId, getKeyHash(key))
 	if result.Err != nil {
 		if result.Err.StatusCode == http.StatusNotFound {
 			return nil, nil
 		}
 		mlog.Error(result.Err.Error())
 		return nil, result.Err
 	}
```

Comment:
If we didn't return at this point yet, we should update the hashed key with the normal key in the database, to not only phase the hashed versions out via the `Set` method, but also via `Get`. Considering we got all information at line 67, we would only need to call effectively `SetPluginKeyWithExpiry`, copying all values from the function call / the returned KV Value from the database fetch. Combined with the requested change from the `SetPluginKeyWithExpiry` that will then 1. create the new version and 2. clean up the old value.